Meta/CMake/Superbuild/CMakeLists.txt

```diff
@@ -44,6 +44,14 @@ if(NOT SERENITY_TOOLCHAIN STREQUAL "GNU")
 endif()
 set(SERENITY_BUILD_DIR "${PROJECT_BINARY_DIR}/../${SERENITY_ARCH}${SERENITY_BUILD_DIR_SUFFIX}")
 
+# When the path stem is outside of the sysroot pkgconf ignores it.
+# Releases of pkgconf shipped by distros incorrectly resolve paths
+# making it believe that the path stem is outside of the sysroot.
+get_filename_component(
+    SERENITY_BUILD_DIR "${SERENITY_BUILD_DIR}"
+    ABSOLUTE
+)
```

Comment:
All of this easily fits into three lines, there is no need to wrap code and comments at 65 characters.

Reply:
I wrote out `get_filename_component` this way to keep it the same with the only other call to it in the same file.

https://github.com/SerenityOS/serenity/blob/db2e1bfa027774ff1e1f883c4ddadc277c1e6e1e/Meta/CMake/Superbuild/CMakeLists.txt#L31-L34

I am unhappy with the current comments and am unsure how to summarize the issue in the best way.
In the next rebase I'll summarize the issue at hand a bit more in the first comment and be a bit more specific in the second.

Reply:
Maybe something like the following?

> pkgconf incorrectly discards a sysroot if it doesn't match the start of the path to the library file. To avoid that, resolve our sysroot into an absolute and canonical path that matches pkgconf's result for resolving the library file.

That would take up three lines of comment (if wrapped at a reasonable 90 characters), but I feel like the extra information is worth it.

PS: Looking at the linked pkgconf commit again, it seems like that wasn't the entire fix for the issue at hand though, right? It may not break as hard now, but the general issue of "library path gets resolved but sysroot path doesn't, and that will cause it to mismatch" still exists.

Reply:
> Maybe something like the following?
> 
> > pkgconf incorrectly discards a sysroot if it doesn't match the start of the path to the library file. To avoid that, resolve our sysroot into an absolute and canonical path that matches pkgconf's result for resolving the library file.
> 
> That would take up three lines of comment (if wrapped at a reasonable 90 characters), but I feel like the extra information is worth it.

Looks good to me, will push in a minute.

> PS: Looking at the linked pkgconf commit again, it seems like that wasn't the entire fix for the issue at hand though, right?

It was the commit where the issue was fixed when bisecting it.

I tested it by running
```bash
export PKG_CONFIG_SYSROOT_DIR="/path/to/serenity/Build/superbuild-x86_64/../x86_64/Root"
export PKG_CONFIG_LIBDIR="/path/to/serenity/Build/x86_64/Root/usr/local/lib/pkgconfig"

pkg-config --libs libpcre2-8
```

on that commit (a61193c7236f5b240585c4f8eef6f452f1d9a7ee) it returns the correct
`-L/path/to/serenity/Build/superbuild-x86_64/../x86_64/Root/usr/local/lib -lpcre2-8`

while a commit before (d68a867dc0297b51925d7d4ecc6704b147fd7116) it returns
`-L/usr/local/lib -lpcre2-8`

> It may not break as hard now, but the general issue of "library path gets resolved but sysroot path doesn't, and that will cause it to mismatch" still exists.

Its not that the library path gets resolved, that is something meson appears to be doing.
pkgconf simply couldn't tell that two paths pointed to the same folder using different methods so it ended up believing the library path was outside of the sysroot.